src/lib.rs

```diff
@@ -30,31 +30,26 @@
 //! ## Finding all occurrences of keys in an input text
 //!
 //! To search for all occurrences of registered keys in an input text,
-//! use [`Trie::common_prefix_searcher()`] for all starting positions in the text.
+//! use [`Trie::common_prefix_search()`] for all starting positions in the text.
 //!
 //! ```
 //! use crawdad::Trie;
 //!
 //! let keys = vec!["世界", "世界中", "国民"];
 //! let trie = Trie::from_keys(&keys).unwrap();
 //!
-//! let mut searcher = trie.common_prefix_searcher();
-//! searcher.update_haystack("国民が世界中にて".chars());
-//!
+//! let haystack: Vec<_> = "国民が世界中にて".chars().collect();
```

Comment:
This is the documentation, so full type annotations are more helpful.
```suggestion
//! let haystack: Vec<char> = "国民が世界中にて".chars().collect();
```